lib/happo/uploader.rb

```diff
@@ -18,6 +19,9 @@ def upload_diffs
 
       bucket = find_or_build_bucket
       dir = SecureRandom.uuid
+      if !(@s3_bucket_path.nil? || @s3_bucket_path.empty?)
+        dir = File.join(@s3_bucket_path, SecureRandom.uuid)
+      end
```

Comment:
We like for our commits to be small and atomic. It would have been nice to keep this change that adds the S3_BUCKET_PATH in a separate commit. Don't worry about splitting it up now, but please keep it in mind for next time.